compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -100,9 +102,32 @@ enum class VectorPreProcStrategy {
   // be masked-out.
   Masking,
   // Do not apply any vectorization pre-processing transformation.
-  None
+  None,
+  // A hint for the compiler to use its heuristics to determine an
+  // actual pre-processing strategy.
+  Heuristics
 };
 
+static llvm::cl::opt<VectorPreProcStrategy> clPProcStrategy(
+    "iree-codegen-llvmcpu-vector-pproc-strategy",
+    llvm::cl::desc("Set the strategy for pre-processing Linalg operation "
+                   "before vectorization:"),
+    llvm::cl::values(
+        clEnumValN(VectorPreProcStrategy::Peeling, "peel",
+                   "Peel iterations from the vector dimensions so that they "
+                   "become multiple of the vector length"),
+        clEnumValN(
+            VectorPreProcStrategy::Masking, "mask",
+            " Compute vector dimensions assuming vector masking support. "
+            "Vector sizes may be rounded up to the nearest power of two "
+            "and out-of-bounds elements would be masked-out."),
+        clEnumValN(
+            VectorPreProcStrategy::None, "none",
+            "Do not apply any vectorization pre-processing transformation."),
+        clEnumValN(VectorPreProcStrategy::Heuristics, "heurystics",
```

Comment:
nit: here is one more

```suggestion
        clEnumValN(VectorPreProcStrategy::Heuristics, "heuristics",
```